src/modules/0.0.7/aws_acm_list/index.ts

```diff
@@ -1,21 +1,46 @@
+import { CertificateDetail, } from '@aws-sdk/client-acm'
+
 import { AWS, } from '../../../services/gateways/aws'
 import { Context, Crud2, Mapper2, Module2, } from '../../interfaces'
-import { Certificate } from './entity'
+import {
+  Certificate,
+  certificateTypeEnum,
+  certificateRenewalEligibilityEnum,
+  certificateStatusEnum,
+} from './entity'
 import * as metadata from './module.json'
 
+// Get Typescript to understand enum matching
+const isCertType = (t?: string): t is certificateTypeEnum => {
+  if (!t) return false;
+  return Object.values<string>(certificateTypeEnum).includes(t);
+}
+const isRenewalEligibility = (t?: string): t is certificateRenewalEligibilityEnum => {
+  if (!t) return false;
+  return Object.values<string>(certificateRenewalEligibilityEnum).includes(t);
+}
+const isStatusType = (t?: string): t is certificateStatusEnum => {
+  if (!t) return false;
+  return Object.values<string>(certificateStatusEnum).includes(t);
+}
+
 export const AwsAcmListModule: Module2 = new Module2({
   ...metadata,
   utils: {
-    certificateMapper: (e: any) => {
+    certificateMapper: (e: CertificateDetail) => {
       const out = new Certificate();
-      if (!e?.CertificateArn) throw new Error('No CertificateArn defined');
+      // To ignore faulty data in AWS, instead of throwing an error on bad data, we return undefined
+      // if (!e?.CertificateArn) throw new Error('No CertificateArn defined');
+      if (!e.CertificateArn) return undefined;
       out.arn = e.CertificateArn;
       out.certificateId = e.CertificateArn.split('/').pop();
-      out.certificateType = e.Type;
+      if (isCertType(e.Type)) out.certificateType = e.Type;
```

Comment:
I guess this is needed instead of just `out.certificateType = e.Type as certificateTypeEnum;` because `e.Type` could be `null`/`undefined` or there is another reason?

Reply:
No the new `isCertType` function returns true when the string is that enum type.

If I had to make that change my current PR wouldn't compile and the common test suite would fail.

Reply:
ok, so checking that unexpected strings are not being assigned, got it